plugins/cpp/parser/src/doccommentformatter.cpp

```diff
@@ -280,166 +253,135 @@ void CommentToHTMLConverter::visitInlineCommandComment(
   {
     case InlineCommandComment::RenderNormal:
       for (unsigned i = 0, e = c_->getNumArgs(); i != e; ++i)
-      {
-        appendToResultWithHTMLEscaping(c_->getArgText(i));
-        _res << " ";
-      }
+        _res << c_->getArgText(i).str() << ' ';
       return;
 
     case InlineCommandComment::RenderBold:
       assert(c_->getNumArgs() == 1);
-      _res << "<b>";
-      appendToResultWithHTMLEscaping(arg0);
-      _res << "</b>";
+      _res << "**" << arg0.str() << "**";
       return;
 
     case InlineCommandComment::RenderMonospaced:
       assert(c_->getNumArgs() == 1);
-      _res << "<tt>";
-      appendToResultWithHTMLEscaping(arg0);
-      _res<< "</tt>";
+      _res << '`' << arg0.str() << '`';
       return;
 
     case InlineCommandComment::RenderEmphasized:
       assert(c_->getNumArgs() == 1);
-      _res << "<em>";
-      appendToResultWithHTMLEscaping(arg0);
-      _res << "</em>";
+      _res << '*' << arg0.str() << '*';
       return;
   }
 }
 
-void CommentToHTMLConverter::visitHTMLStartTagComment(
+void CommentToMarkdownConverter::visitHTMLStartTagComment(
   const HTMLStartTagComment* c_)
 {
-  _res << "<" << c_->getTagName().str();
+  _res << '<' << c_->getTagName().str();
 
   if (c_->getNumAttrs())
     for (unsigned i = 0, e = c_->getNumAttrs(); i != e; i++)
     {
-      _res << " ";
+      _res << ' ';
       const HTMLStartTagComment::Attribute &Attr = c_->getAttr(i);
```

Comment:
There's still a reference to some "HTML" stuff here.

Reply:
I don't know why GitHub shows you " instead of ', but for me it is '.
`HTMLStartTagComment` is in `clang::comments`, it is not ours. This describes HTML fragments embedded in doc comments.

Reply:
Why is there a `visitHTMLStartTagComment` and a `visitHTMLEndTagComment` method? Are they required?

Reply:
Parts of doc comments are visited by a clang visitor. Clang visits the `\brief` part, the documentation of parameters, HTML tags, etc.
```.cpp
/**
 * \brief This is a brief description.
 *
 * This is a <b>long description</b> of this smart function containing <i>HTML</i> tags.
 * \param x This is the first parameter.
 * \param y This is the second parameter.
 * \return The function returns an integer.
 */
int func(int x, int y);
```
So in this example the brief, the long description, parameter and return value documentation are visited separately by clang. Also `<b>` and `<i>` tags are visited by `visitHTMLStartTagComment` visitor function. To be honest, I haven't tried if these tags are present in their original form as the content of main description. It this case we really shouldn't visit them separately. I'll check it later.

Reply:
> I don't know why GitHub shows you " instead of ', but for me it is '.
@bruntib My comment is tagging line 321/285, which is one line below your changes. I can see the changed code appending a single char as a char and not as a string literal. :slightly_smiling_face: I put "HTML" into quotes to indicate that there are words in the program that still refer to HTML stuff.

Reply:
Oh, I got it. I misunderstood this. So the HTML stuff here is conscious and it really refers to HTML things, but I'll check if we really need it.

Reply:
Well, *Doxygen* in the standard format generates HTML and thus I guess it is allowed (in the source code) so that you, as the library developer, can embed HTML formatting to your generated documentation... I'm interesting what Clang does with this in the first place...

Reply:
Yes, that's the embedded HTML that this visitor function processes. I've checked it and the result markdown wouldn't contain these HTML tags if these visitor functions are missing. So I consider this as the final solution.